### ISSUE TITLE
Inconsistent API for stopping a connection in progress

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7464]
|<img alt="dizzyd's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| dizzyd reported|
|-|-|
|Trac ID|trac#115|
|Type|defect|
|Created|2003-07-23 04:02:11Z|

<details><summary>Searchable metadata</summary>

```
trac-id__115 115
type__defect defect
reporter__dizzyd dizzyd
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1058932931000000 1058932931000000
changetime__1465027924017082 1465027924017082
version__ 
owner__ 
cc__glyph cc__itamarst cc__jml cc__dizzyd cc__moshez
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="dizzyd's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|dizzyd commented|
|-|-|

```
#!html
<pre>
We need to codify the API such that stopping a connection 
is consistent, regardless if the connection is in progress 
or not. I.e. drop stopConnecting() and just have 
loseConnection() do the Right Thing.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Find somebody who can fix this properly? :)

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Reassigning because moshez refuses to use the bugtracker. 
Sorry :(

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
Sorry, going away for a bit.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
It's spelled "disconnect", and is in the IConnector docstrings and
seems to be implemented correctly. If you actually have an
issue with disconnect not working, please open a new bug
and specify the exact circumstances.

Resolving this bug now.
Thanks
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

In changeset 8ebcec5473d6c716ff3278c72f6a17b6b90f64ea
```
#!CommitTicketReference repository="" revision="8ebcec5473d6c716ff3278c72f6a17b6b90f64ea"
Merge pull request #115 from rodrigc/8399-rodrigc-print: Convert from print statement to print() function in twisted/scripts

Author: rodrigc
Reviewer: cyli
Fixes: #8399

Convert from print statement to print() function in twisted/scripts
```
